_assets/js/click-tracking.js

```diff
@@ -0,0 +1,43 @@
+// Import our constants from constandts.js
+import { CLICK_TRACKING_ENDPOINT, ACCESS_KEY, AFFILIATE, MODULE_CODE } from './utils/constants';
+
+export default function clickTracking() {
+  // Get all the anchor tags that are children of b tags on the page:
+  const allLinks = document.querySelectorAll('.search-result');
+
+  // Loop through all the links and add a click event listener to each one:
+  allLinks.forEach((link, index) => {
+    const url = link.href;
+    const query = location.search.split('&').find((param) => param.startsWith('query='));
+    const position = location.search.split('&').includes('offset=') ?
+    Number(
+        location.search
+          .split('&')
+          .find((param) => param.startsWith('offset='))
```

Comment:
For `param.startsWith('offset=')`, what happens if offset is the first param? 

e.g., `/some/route?offset=12&query=10`


I think a generalized `function getParams()` might be useful here, since I see we're splitting on '&' in a few places to achieve this.

Modern browsers have [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/get#examples), but we could maybe implement something that takes `location.search` and returns a `Map([[paramOne, valueOne], [paramTwo, valueTwo]])` etc?

Reply:
So I actually already have a function called getSearchParams already written, 🤦🏻 

The way it currently works is it takes the param you want to get, like query or offset, and returns the value in a string. Since it's a string, it does need to be URI escaped. Are you leaning towards using a map for any particular reasons?

Reply:
Nah - I suggested a Map because it conforms more closely to URLSearchParams, but just grabbing each param from the string works, too.